rero_ils/modules/contributions/sync.py

```diff
@@ -0,0 +1,546 @@
+                            f'{new_mef_pid} for {source} (pid:{old_agent_pid})'
+                        )
+                        # if the MEF record does not exists create it
+                        new_agent = Contribution.get_record_by_pid(new_mef_pid)
+
+                        if not new_agent:
+                            if not self.dry_run:
+                                Contribution.create(
+                                    data=new_mef_data,
+                                    dbcommit=True,
+                                    reindex=True
+                                )
+                            self.logger.info(
+                                f'Create a new MEF record(pid: {new_mef_pid})')
+                        else:
+                            # update the new MEF
+                            # recursion
+                            self.logger.info(
+                                f'MEF(pid: {agent.pid}) recursion with'
+                                f' (pid:{new_mef_pid})')
+                            new_doc_updated, new_updated, new_error = \
+                                self.sync_record(new_mef_pid)
+                            # TODO: find a better way
+                            doc_updated.update(new_doc_updated)
+                            updated = updated or new_updated
+                            error = error or new_error
+                    # something changed
+                    # update the content
+                    self.logger.info(
+                        f'MEF(pid: {agent.pid}) content has been '
+                        f'updated')
+                    if not self.dry_run:
+                        if (old_mef_pid == new_mef_pid):
+                            Contribution.get_record_by_id(
+                                agent.id).replace(
+                                    new_mef_data, dbcommit=True, reindex=True)
+                        else:
+                            # as we have only the last mef but not the old one
+                            # we need get it from the MEF server
+                            # this is important as it can still used by other
+                            # agents
+                            Contribution.get_record_by_pid(
+                                pid).update_online(dbcommit=True, reindex=True)
+                    updated = True
+
+            if updated:
+                # for each documents
+                self.logger.info(
+                    f'MEF(pid: {agent.pid}) try to update '
+                    f'documents: {doc_pids}')
+                for doc_pid in doc_pids:
+                    doc_is_updated = self.update_agents_in_document(
+                        doc_pid=doc_pid,
+                        pids_to_replace=pids_to_replace
+                    )
+                    if doc_is_updated:
+                        doc_updated.add(doc_pid)
```

Comment:
`self.update_agents_in_document` can only return `True` (maybe it could raise an exception) so :
```python
for pid in docs_pids:
    self.update_agents_in_document(doc_pid=pid, pids_to_replace=pids_to_replace)
....
return doc_pids, updated, error
```